tests/unit/amo/pages/TestAddonInfo.js

```diff
@@ -70,82 +77,351 @@ describe(__filename, () => {
+      });
+
+      sinon.assert.calledWith(
+        dispatch,
+        fetchAddon({
+          errorHandler,
+          slug,
+        }),
+      );
+
+      sinon.assert.calledWith(
+        dispatch,
+        fetchAddonInfo({
+          errorHandlerId: errorHandler.id,
+          slug,
+        }),
+      );
+    },
+  );
+
+  it.each([ADDON_INFO_TYPE_EULA, ADDON_INFO_TYPE_PRIVACY_POLICY])(
+    `fetches an addon and addonInfo for %s when the slug changes`,
+    (infoType) => {
+      const slug = 'some-slug';
+      const newSlug = 'some-other-slug';
+      const addon = { ...fakeAddon, slug };
+      _loadAddonResults([addon]);
+      const dispatch = sinon.stub(store, 'dispatch');
+      const errorHandler = createStubErrorHandler();
+      render({
+        errorHandler,
+        params: { slug },
+      });
+
+      dispatch.resetHistory();
+      render({
+        errorHandler,
+        infoType,
+        params: { slug: newSlug },
+      });
+
+      sinon.assert.calledWith(
+        dispatch,
+        fetchAddon({
+          errorHandler,
+          slug: newSlug,
+        }),
+      );
+
+      sinon.assert.calledWith(
+        dispatch,
+        fetchAddonInfo({
+          errorHandlerId: errorHandler.id,
+          slug: newSlug,
+        }),
+      );
+    },
+  );
+
+  it.each([ADDON_INFO_TYPE_EULA, ADDON_INFO_TYPE_PRIVACY_POLICY])(
+    `does not fetch addonInfo for %s if it is already loaded`,
+    (infoType) => {
+      const slug = 'some-addon-slug';
+      _loadAddonInfo({ slug });
+      const errorHandler = createStubErrorHandler();
+
+      const fakeDispatch = sinon.stub(store, 'dispatch');
+
+      render({
+        errorHandler,
+        infoType,
+        params: { slug },
+      });
+
+      sinon.assert.neverCalledWith(
+        fakeDispatch,
+        fetchAddonInfo({
+          errorHandlerId: errorHandler.id,
+          slug,
+        }),
+      );
+    },
+  );
+
+  it.each([ADDON_INFO_TYPE_EULA, ADDON_INFO_TYPE_PRIVACY_POLICY])(
+    `does not fetch addonInfo for %s if it is already loading`,
+    (infoType) => {
+      const slug = 'some-addon-slug';
+      const errorHandler = createStubErrorHandler();
+
+      store.dispatch(fetchAddonInfo({ errorHandlerId: errorHandler.id, slug }));
+
+      const fakeDispatch = sinon.stub(store, 'dispatch');
+
+      render({
+        errorHandler,
+        infoType,
+        params: { slug },
+      });
+
+      sinon.assert.neverCalledWith(
+        fakeDispatch,
+        fetchAddonInfo({
+          errorHandlerId: errorHandler.id,
+          slug,
+        }),
+      );
+    },
+  );
+
+  describe('ADDON_INFO_TYPE_CUSTOM_LICENSE', () => {
+    it('fetches an addon when requested by slug', () => {
+      const slug = 'some-addon-slug';
+      const dispatch = sinon.stub(store, 'dispatch');
+      const errorHandler = createStubErrorHandler();
+
+      render({
+        errorHandler,
+        infoType: ADDON_INFO_TYPE_CUSTOM_LICENSE,
+        params: { slug },
```

Comment:
I see you use this a few times so I am wondering to make this a little bit more DRY there could be a `renderLicense`? not sure if it's worth it...there may be other ways too

Reply:
Good idea! I'll do that.